Browser_IDE/executionEnvironment_Internal.js

```diff
@@ -215,32 +215,35 @@ ReferenceError: test is not defined
 // Currently those are the only two forms supported, but this should account for the majority well enough.
 // It also doesn't parse the url style ones - it only needs to work for the local user's code, so good enough.
 
+
+
 function parseErrorStack(err){
     const stackParse = /(?:@|\()((?:[^;:`]|[:;`](?=.*(?:\/|\.)))*)`?;?:([0-9]*)/g;
     let stack = [...err.stack.matchAll(stackParse)];
 
     let stackIndex = 0;
 
-    //Should we limit this to only SplashKitArgumentError? (i.e if (err instanceof SplashKitArgumentError))
-
     // Unwind stack until we find user code:
     while(stackIndex < stack.length && !stack[stackIndex][1].startsWith(userCodeBlockIdentifier))
         stackIndex += 1;
 
-    if (stackIndex >= stack.length)
-        stackIndex = 0;
-
-    let lineNumber = stack[stackIndex][2];
+    // Only include the stack frames that are relevant to your code
+    stack = stack.slice(stackIndex, stackIndex + 2);
 
-    let file = stack[stackIndex][1];
+    let lineNumber = stack[0][2];
+    let file = stack[0][1];
 
     if (file.startsWith(userCodeBlockIdentifier))
         lineNumber -= userCodeStartLineOffset;
 
-    return {lineNumber, file};
+    // Include the error message at the beginning of the stack trace
+    stack.unshift([err.message]);
+    stack = stack.map(s=>s.join(':')).join('\n');
+    return {lineNumber, file, stack};
```

Comment:
It'd be nice to improve the formatting here a bit - currently when I test it it looks like this:
![image](https://github.com/thoth-tech/SplashkitOnline/assets/42032199/03ae8167-fa9a-42ab-a849-438b5d685675)

It'd be nicer perhaps if it was like this?
![image](https://github.com/thoth-tech/SplashkitOnline/assets/42032199/1a7b3fc4-d645-4ecb-9c9a-52139e3e247e)
_(just to clarify, this screenshot is also with a fix for the hard-coded two-line limit as well, hence the extra line)_

Important points are:
1. no need to output the first part of each stack line, since the last two parts contain the same information
2. line numbers need to have `userCodeStartLineOffset` subtracted from them or they aren't correct (make sure to check the output!)
3. make sure to slice off the `userCodeBlockIdentifier`

Reply:
Sorry this was meant to be part of the main review I left, accidentally clicked the wrong button 😅